### PR TITLE
TC_04.001.03 | Folder > Rename Folder | Verify that an error message is displayed when an invalid folder name is entered in the Rename Folder field 

### PR DESCRIPTION
**Pull request by Ridgewood2016**
https://github.com/RedRoverSchool/JenkinsQA_JS_2024_fall/issues/25

### CODE REVIEW

**Comment by Ridgewood2016** [comment]
https://github.com/RedRoverSchool/JenkinsQA_JS_2024_fall/issues/25

File: cypress/e2e/folderRenameFolder.cy.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/// <reference types="cypress"/>
+const USERNAME = Cypress.env('local.admin.username');
+const PASSWORD = Cypress.env('local.admin.password');
```
Comment (Negrescu29) [inline cypress/e2e/folderRenameFolder.cy.js:3]: These 2 variables (USERNAME, PASSWORD) are not used in this test, better to remove.

**Review by Negrescu29 — CHANGES_REQUESTED** [review]

File: cypress/e2e/folderRenameFolder.cy.js (hunk 1 above)
Comment (Ridgewood2016) [reply cypress/e2e/folderRenameFolder.cy.js]: @Negrescu29  Fixed, please approve

**Review by Negrescu29 — APPROVED** [review]
👍🏼

File: cypress/e2e/folderRenameFolder.cy.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/// <reference types="cypress"/>
+const LOCAL_PORT = Cypress.env('local.port');
+const LOCAL_HOST = Cypress.env('local.host');
+
+describe('US_04.001 |Folder > Rename Folder', () => {
+        it('Verify that error message is displayed when an invalid folder name is entered in the Rename Folder field', () => {
+            cy.get('span').contains('New Item').click()
+            cy.wait(1000)
```
Comment (adiuzhikov) [inline cypress/e2e/folderRenameFolder.cy.js:8]: Is it necessary to use .wait() here?

File: cypress/e2e/folderRenameFolder.cy.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            cy.get('input[name="name"]').clear()
```
Comment (adiuzhikov) [inline cypress/e2e/folderRenameFolder.cy.js:9]: Why do you need to .clear() the input?

File: cypress/e2e/folderRenameFolder.cy.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            cy.get('button').contains('Save').click()
+            cy.get('#jenkins-home-link').click()
+            cy.get('[href="job/Folder1/"]').realHover({ position: 'center' })
+            cy.get(`[data-href="http://${LOCAL_HOST}:${LOCAL_PORT}/job/Folder1/"]`).should('be.visible')
+            cy.get(`[data-href="http://${LOCAL_HOST}:${LOCAL_PORT}/job/Folder1/"]`).click();
+            cy.get('a[href="/job/Folder1/confirm-rename"]').click()
+            cy.get('input[name="newName"]').clear()
+            cy.get('input[name="newName"]').type('Folder*')
+            cy.get('button[name="Submit"]').click()
+            cy.get('div[id="main-panel"]').contains('Error')
```
Comment (adiuzhikov) [inline cypress/e2e/folderRenameFolder.cy.js:22]: Looks like your test doesn't have an assertion - so what do you check here?

**Comment by Ridgewood2016** [comment]
@adiuzhikov Fixed. Please approve

**Comment by adiuzhikov** [comment]
You still don't have an assertion in the test. Please, use .should() or .expect() instead of .contains()

**Comment by Ridgewood2016** [comment]
@adiuzhikov . fixed. Please check.

**Comment by Ridgewood2016** [comment]
Need to rework

**Pull request closed without merging**